fdbserver/VersionedBTree.actor.cpp

```diff
@@ -1286,108 +1289,241 @@ int nextPowerOf2(uint32_t x) {
 
 struct RedwoodMetrics {
 	static constexpr int btreeLevels = 5;
+	static int maxRecordCount;
 
-	RedwoodMetrics() { clear(); }
+	struct eventReasonsArray{
+		unsigned int eventReasons[(size_t)events::MAXEVENTS][(size_t)pagerEventReasons::MAXEVENTREASONS];
 
-	void clear() {
-		memset(this, 0, sizeof(RedwoodMetrics));
-		for (auto& level : levels) {
-			level = {};
+		eventReasonsArray(){clear();}
+		void clear(){
+			for(size_t i = 0; i<(size_t)events::MAXEVENTS; i++){
+				for(size_t j = 0; j<(size_t)pagerEventReasons::MAXEVENTREASONS; j++){
+					eventReasons[i][j] = 0;
+				}
+			}
 		}
-		startTime = g_network ? now() : 0;
-	}
+		void addEventReason(events e, pagerEventReasons r){
+			eventReasons[(size_t)e][(size_t)r] += 1;
+		}
+		const unsigned int& getEventReason(events e, pagerEventReasons r){
+			return eventReasons[(size_t)e][(size_t)r];
+		}
+
+		std::string ouputSummary(int currLevel){
+			std::string result = "";
+			std::map<events, std::string> allEvents = {{events::pagerCacheLookup, "pagerCacheLookup"}, {events::pagerCacheHit, "pagerCacheHit"}, {events::pagerCacheMiss, "pagerCacheMiss"}, {events::pagerWrite, "pagerWrite"}};
```

Comment:
Instead of a map, just use `static const std::pair<events, const char *>[] = ...`

Iteration is the same interface, but it doesn't build a map so the initialization order is preserved.  Overhead here does not particularly matter but it's also much less overhead.